### DISCUSSION TITLE
How to change AMF and OP?

### DISCUSSION THREAD

**Discussion by giscard85** [magma/magma#15271] [Q&A]
Dear Team,
I setup magma in order to explore features and test. I'm using physical eNB and UE with commercial SIM card. Orhestrator and AGW are in different VM and working well.
With default AMF, when I add new IMSI with my opc and K values, some IMSI works but other not. They are different manufacturer but have been make with same encryption values. The same values (OPC, AMF, KI) works well with another commercial EPC for both IMSI.
I changed default value of AMF in orc8r but I don't see where to change OP. After change AMF none of the SIM success to attach.
I can see OP value in /var/opt/magma/configs/gateway.mconfig but I don't see where to change in orchestrator. 
Please help.

Here is MME Logs with new AMF
200010001001733 failed to attach
200010001010264 failed to attach
-------------------------------------------------------------------------------------------------------------------------
Jul 22 12:31:45 magma-dev mme[524992]: [ERROR] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001010264; Status: Connect Failed; ErrorCode: 0
Jul 22 12:31:45 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject
Jul 22 12:31:56 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001010264
Jul 22 12:31:56 magma-dev mme[524992]: [ERROR] 14: Connect Failed
Jul 22 12:31:56 magma-dev mme[524992]: [ERROR] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001010264; Status: Connect Failed; ErrorCode: 0
Jul 22 12:31:56 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject
Jul 22 12:32:07 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001010264
Jul 22 12:32:10 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject
Jul 22 12:32:15 magma-dev mme[524992]: [ERROR] 14: Connect Failed
Jul 22 12:32:15 magma-dev mme[524992]: [ERROR] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001010264; Status: Connect Failed; ErrorCode: 0
Jul 22 12:33:25 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001010264
Jul 22 12:33:29 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject
Jul 22 12:33:30 magma-dev mme[524992]: [ERROR] 14: Connect Failed
Jul 22 12:33:30 magma-dev mme[524992]: [ERROR] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001010264; Status: Connect Failed; ErrorCode: 0
Jul 22 12:33:39 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001010264
Jul 22 12:33:42 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject
Jul 22 12:33:43 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001001733
Jul 22 12:33:46 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject
Jul 22 12:33:49 magma-dev mme[524992]: [ERROR] 4: Deadline Exceeded
Jul 22 12:33:49 magma-dev mme[524992]: [ERROR] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001010264; Status: Deadline Exceeded; ErrorCode: 0
Jul 22 12:33:52 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001010264
Jul 22 12:33:53 magma-dev mme[524992]: [ERROR] 4: Deadline Exceeded
Jul 22 12:33:53 magma-dev mme[524992]: [ERROR] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001001733; Status: Deadline Exceeded; ErrorCode: 0
Jul 22 12:33:55 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject
Jul 22 12:33:56 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001001733
Jul 22 12:33:59 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject
Jul 22 12:34:02 magma-dev mme[524992]: [ERROR] 4: Deadline Exceeded
Jul 22 12:34:02 magma-dev mme[524992]: [ERROR] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001010264; Status: Deadline Exceeded; ErrorCode: 0
Jul 22 12:34:05 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001010264
Jul 22 12:34:06 magma-dev mme[524992]: [ERROR] 4: Deadline Exceeded
Jul 22 12:34:06 magma-dev mme[524992]: [ERROR] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001001733; Status: Deadline Exceeded; ErrorCode: 0
Jul 22 12:34:08 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject
Jul 22 12:34:09 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001001733
Jul 22 12:34:12 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject
Jul 22 12:34:15 magma-dev mme[524992]: [ERROR] 4: Deadline Exceeded
Jul 22 12:34:15 magma-dev mme[524992]: [ERROR] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001010264; Status: Deadline Exceeded; ErrorCode: 0
Jul 22 12:34:18 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001010264
Jul 22 12:34:19 magma-dev mme[524992]: [ERROR] 4: Deadline Exceeded
Jul 22 12:34:19 magma-dev mme[524992]: [ERROR] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001001733; Status: Deadline Exceeded; ErrorCode: 0
Jul 22 12:34:21 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject
Jul 22 12:34:22 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001001733
Jul 22 12:34:25 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject
Jul 22 12:34:29 magma-dev mme[524992]: [ERROR] 4: Deadline Exceeded
Jul 22 12:34:29 magma-dev mme[524992]: [ERROR] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001010264; Status: Deadline Exceeded; ErrorCode: 0
Jul 22 12:34:32 magma-dev mme[524992]: [ERROR] 4: Deadline Exceeded
Jul 22 12:34:32 magma-dev mme[524992]: [ERROR] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001001733; Status: Deadline Exceeded; ErrorCode: 0
Jul 22 12:34:36 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001001733
Jul 22 12:34:39 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject
Jul 22 12:34:46 magma-dev mme[524992]: [ERROR] 4: Deadline Exceeded
Jul 22 12:34:46 magma-dev mme[524992]: [ERROR] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001001733; Status: Deadline Exceeded; ErrorCode: 0

MME Logs with default AMF
200010001001733 attach successfully
200010001010264 failed to attach
-------------------------------------------------------------------------------------------------------------------------
Jul 22 12:41:41 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001001733
Jul 22 12:41:41 magma-dev mme[524992]: [INFO] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001001733; Status: ; StatusCode: 2001
Jul 22 12:41:41 magma-dev mme[524992]: [DEBUG] Sending S6A-UPDATE_LOCATION_REQUEST with IMSI: 200010001001733
Jul 22 12:41:41 magma-dev mme[524992]: [INFO] Received S6A-LOCATION-UPDATE_ANSWER for IMSI: 200010001001733; Status: ; StatusCode: 2001
Jul 22 12:41:41 magma-dev mme[524992]: [INFO] sent itti S6A-LOCATION-UPDATE_ANSWER for IMSI: 200010001001733
Jul 22 12:41:42 magma-dev mme[524992]: /home/vagrant/magma/lte/gateway/c/core/oai/tasks/sgw/sgw_handlers.cpp:403:11: runtime error: null pointer passed as argument 1, which is declared to never be null
Jul 22 12:41:42 magma-dev mme[524992]: /home/vagrant/magma/lte/gateway/c/core/oai/tasks/sgw/sgw_handlers.cpp:739:11: runtime error: null pointer passed as argument 1, which is declared to never be null
Jul 22 12:41:42 magma-dev sudo[532504]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/usr/local/bin/magma-create-gtp-port.sh g_ad410ac 172.16.212.10 true false false
Jul 22 12:41:42 magma-dev sudo[532504]: pam_unix(sudo:session): session opened for user root by (uid=0)
Jul 22 12:41:42 magma-dev sudo[532507]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/usr/bin/ovs-vsctl --may-exist add-port gtp_br0 g_ad410ac -- set Interface g_ad410ac type=gtpu options:remote_ip=172.16.212.10 options:key=flow bfd:enable=true options:csum=false bfd:min_tx=5000 bfd:min_rx=5000
Jul 22 12:41:42 magma-dev sudo[532507]: pam_unix(sudo:session): session opened for user root by (uid=0)
Jul 22 12:41:42 magma-dev mme[524992]: [DEBUG] Success logging event: attach_success
Jul 22 12:41:42 magma-dev ovs-vsctl[532508]: ovs|00001|vsctl|INFO|Called as ovs-vsctl --may-exist add-port gtp_br0 g_ad410ac -- set Interface g_ad410ac type=gtpu options:remote_ip=172.16.212.10 options:key=flow bfd:enable=true options:csum=false bfd:min_tx=5000 bfd:min_rx=5000
Jul 22 12:41:42 magma-dev sudo[532507]: pam_unix(sudo:session): session closed for user root
Jul 22 12:41:42 magma-dev sudo[532504]: pam_unix(sudo:session): session closed for user root
Jul 22 12:41:42 magma-dev sudo[532510]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/usr/bin/ovsdb-client dump Interface name ofport
Jul 22 12:41:42 magma-dev sudo[532510]: pam_unix(sudo:session): session opened for user root by (uid=0)
Jul 22 12:41:42 magma-dev sudo[532510]: pam_unix(sudo:session): session closed for user root
Jul 22 12:42:46 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001010264
Jul 22 12:42:46 magma-dev mme[524992]: [INFO] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001010264; Status: ; StatusCode: 2001
Jul 22 12:42:46 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001010264
Jul 22 12:42:46 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject
Jul 22 12:42:47 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001010264
Jul 22 12:42:47 magma-dev mme[524992]: [INFO] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 200010001010264; Status: ; StatusCode: 2001
Jul 22 12:42:47 magma-dev mme[524992]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 200010001010264
Jul 22 12:42:50 magma-dev mme[524992]: [DEBUG] Success logging event: attach_reject

Thanks

**Comment by panyogesh** [comment]
@giscard85 : Can you please share the configuration values (both from UE and Magma ) and I can check on that. Also I am assuming you are making 4G calls

**Reply by giscard85** [reply]
Hi @panyogesh,
Thanks for replying.
I'm not making calls, only for Data.
Can you give me which configurations exactly? 
I added IMSI with Orc8r with apn, and configure ue with same apn.
![image](https://github.com/magma/magma/assets/52927668/e8ed092e-1d10-4cb7-9e7f-38d8b41a4b88)

**Comment by panyogesh** [comment]
@giscard85 : We tested both 4G and 5G call and seems to be through.  Can you please share of AMF and OP value.

**Reply by giscard85** [reply]
Hi,
Sorry for late reply.
I'm using only DATA not call.
Those are value for UE
ki: 19e04dda4355eb86ca322d13e05839a4
Op: fa4cffa4bf13e5fec3bc1d7ee8c5d978
AMF: 4c8c8574a5a36acd7c2cfd4c88bbebf0
opc: ffffffffffffffffffffffffffffffff

On server
k: 5C95978B5E89488CB7DB44381E237809
opc: d59b9f864e1feff9c3131783238e11ed
amf: gAA=

If I changed  default AMF on server, none of the SIM can attach.

Regards
Giscard

**Reply by panyogesh** [reply]
@giscard85 : Sorry for the delayed response. Is the issue still observed. If so I can ask one of the test engg. to verify.

**Comment by panyogesh** [comment]
Closing as conversation is bit outdated. Please raise new issue/discussion if its observed on latest builds.
